odoo/addons/base/models/res_currency.py

```diff
@@ -313,11 +313,15 @@ def _get_view(self, view_id=None, view_type='form', **options):
         arch, view = super()._get_view(view_id, view_type, **options)
         if view_type in ('tree', 'form'):
             currency_name = (self.env['res.company'].browse(self._context.get('company_id')) or self.env.company.root_id).currency_id.name
-            for field in [['company_rate', _('Unit per %s', currency_name)],
-                          ['inverse_company_rate', _('%s per Unit', currency_name)]]:
-                node = arch.xpath("//tree//field[@name='%s']" % field[0])
+            fields_maps = [
+                [['company_rate', 'inverse_rate'], _('Unit per %s', currency_name)],
+                [['inverse_company_rate', 'rate'], _('%s per Unit', currency_name)],
```

Comment:
This is going to be confusing indeed :upside_down_face: 

Note that a solution to go back to the way it was in 16.0 would be to invert both of these:
* the computation of `rate` and `inverse_rate` (what you did before)
* use `inverse_rate` instead of `rate` in `_get_conversion_rate` (what you didn't do and caused you to change the assertion values of `test_currency_cache`

If you do that, you won't need to change the views in XML (community and enterprise; but the change in python is great no matter what)

Reply:
Since this is in stable, we also need to add the `.pot` entries.

Did you try my suggestion above to avoid changing the views?

Reply:
@william-andre 
I tried ur suggestion but the tests in TestResCurrency still failed.
I will update the `.pot` files now